code/modules/mining/machine_processing.dm

```diff
@@ -196,117 +62,83 @@
 	density = 1
 	anchored = 1
 	var/obj/machinery/mineral/CONSOLE = null
-	var/ore_gold = 0;
-	var/ore_silver = 0;
-	var/ore_diamond = 0;
-	var/ore_glass = 0;
-	var/ore_plasma = 0;
-	var/ore_uranium = 0;
-	var/ore_iron = 0;
-	var/ore_clown = 0;
-	var/ore_adamantine = 0;
-	var/ore_titanium = 0;
-	var/selected_gold = 0
-	var/selected_silver = 0
-	var/selected_diamond = 0
-	var/selected_glass = 0
-	var/selected_plasma = 0
-	var/selected_uranium = 0
-	var/selected_iron = 0
-	var/selected_clown = 0
-	var/selected_titanium = 0
-	var/on = 0 //0 = off, 1 =... oh you know!
+	var/datum/material_container/materials
+	var/on = FALSE
+	var/selected_material = MAT_METAL
+	var/selected_alloy = null
+	var/datum/research/files
+	var/list/categories = list("Alloys")
+
+/obj/machinery/mineral/processing_unit/Initialize()
+	. = ..()
+	proximity_monitor = new(src, 1)
+	materials = new(src, list(MAT_METAL, MAT_GLASS, MAT_SILVER, MAT_GOLD, MAT_DIAMOND, MAT_PLASMA, MAT_URANIUM, MAT_BANANIUM, MAT_TITANIUM, MAT_BLUESPACE),PROCESSING_UNIT_CAPACITY)
+	files = new /datum/research/smelter(src)
+
+/obj/machinery/mineral/processing_unit/Destroy()
+	qdel(materials)

```

Comment:
in fact really that relationship should be two way, so the console can also null that reference if it gets deleted.